### PR TITLE
Check block's indep hash

### PR DESCRIPTION
**Pull request by janekolszak**
- Block verification #8
- Cleanup deep hash code

### CODE REVIEW

File: src/utils/listener/listener.go [hunk 1]
```diff
@@ -196,6 +198,15 @@ func (self *Listener) monitorBlocks() {
 				// FIXME: Inform downstream something's wrong
 			}
 
+			if !block.IsValid() {
```
Comment (ppedziwiatr) [inline src/utils/listener/listener.go:201]: how exactly will it currently behave? retry-ing to validate the same block from the same peer doesn't make much sense..?

File: src/utils/arweave/block.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+func generateBlockDataSegment(b *Block) []byte {
+	bdsBase := generateBlockDataSegmentBase(b)
+
+	values := []any{
+		bdsBase,
+		fmt.Sprintf("%d", b.Timestamp),
+		fmt.Sprintf("%d", b.LastRetarget),
+
+		// TODO: those 3 params are int64 in string form
```
Comment (ppedziwiatr) [inline src/utils/arweave/block.go:78]: either do or remove todo ;-)

Comment (janekolszak) [reply src/utils/arweave/block.go]: Added type to parse the strings into big.Int

File: src/utils/listener/listener.go (hunk 1 above)
Comment (janekolszak) [reply src/utils/listener/listener.go:201]: This will be handled in this issue: https://github.com/warp-contracts/syncer/issues/9

**Review by ppedziwiatr — APPROVED** [review]

**Pull request merged**